Redesign login screen and password recovery

dasJ opened the pull request:
- Add copyright
- Remove odd looking icon and version
- Add a way to navigate back from password recovery
- Add icons to password recovery fields

MCMic commented on html/class_passwordRecovery.inc:
```diff
@@ -147,10 +147,15 @@ class standAlonePage {
     $smarty->assign('password_img', 'geticon.php?context=status&icon=dialog-password&size=48');
+    $smarty->assign('mail_img', 'geticon.php?context=applications&icon=internet-mail&size=48');
+    $smarty->assign('directory_img', "geticon.php?context=places&icon=network-server&size=48");
+    $smarty->assign('personal_img', "geticon.php?context=types&icon=user&size=48");
```
Not sure of the benefit of putting that here, maybe directly put images paths in the tpl file?

MCMic commented on ihtml/themes/breezy/login.tpl:
```diff
@@ -1,7 +1,6 @@
 <body>
 
   {$php_errors}
-  {include file={filePath file="setup_header.tpl"}}
```
Why?
This removes logo and version if I’m not mistaking?
I get that you move version to the copyright notice, but the logo should be there IMO.

MCMic commented on ihtml/themes/breezy/login.tpl:
```diff
@@ -13,7 +12,7 @@
   <div id="window-titlebar">
     <p>
-      {t}Login screen{/t}
+      {t}FusionDirectory{/t}
```
I get that Login Screen is not a usefull title, but FusionDirectory already appears in other places on the screen. Maybe we should just have «Login» as title?

MCMic commented on ihtml/themes/breezy/login.tpl:
```diff
@@ -49,9 +48,9 @@
-          document.cookie = "gosatest=empty;path=/;expires=Thu, 01-Jan-1970 00:00:01 GMT";
+        document.cookie = "fdtest=empty;path=/";
+        if (document.cookie.indexOf( "fdtest=") > -1 )
+          document.cookie = "fdtest=empty;path=/;expires=Thu, 01-Jan-1970 00:00:01 GMT";
```
Good change

MCMic commented on ihtml/themes/breezy/login.tpl:
```diff
@@ -76,6 +75,10 @@
+<div class="copynotice">
+  &copy; 2002-{$year} <a href="http://www.fusionDirectory.org">{t}The FusionDirectory team{/t}, {$revision}</a>
+</div>
+
```
Maybe this should go into one big translatable string to ease translation for RTL languages.
So: `{t 1=$year 2=$revision}&copy; 2002-%1 <a href="http://www.fusiondirectory.org">The FusionDirectory team, %2</a>{/t}`

MCMic commented on ihtml/themes/breezy/recovery.tpl:
```diff
@@ -1,9 +1,6 @@
   {$php_errors}
-  <div>
-    {include file={filePath file="setup_header.tpl"}}
-  </div>
```
Same remark that for index, I disagree about removing the header bar.

MCMic commented on ihtml/themes/breezy/recovery.tpl:
```diff
@@ -40,39 +37,35 @@
+        {html_options options=$server_options selected=$server_id}
+      </select>
+    </div>
+    {/if}
```
I will have to test these changes live, but if I understand them correctly they get rid of table to use divs and they make recovery process close to the index login form, which is not a bad idea as recovery is a public page it might as well match style from login which is the other public page.

MCMic commented on ihtml/themes/breezy/recovery.tpl:
```diff
@@ -85,7 +78,7 @@
     <div class="success">
       <img class="center" src="geticon.php?context=status&amp;icon=task-complete&amp;size=16" alt="{t}Success{/t}" title="{t}Success{/t}">&nbsp;<b>{t}Your password has been changed successfully.{/t}</b><br/>
-      <br/><a href="./">Return to login screen</a>
+      <br/><a href="index.php">{t}Return to login screen{/t}</a>
```
Nice catch about this not being translated

dasJ commented:
I removed the `setup_header.tpl` because both icon and version number looked misplaced as they were just somewhere on the screen without anything near. As you have said, the version is now in the copyright, but I also agree in leaving the FusionDirectory somewhere on the page. Maybe we can put it above the login, like [phpMyAdmin](https://demo.phpmyadmin.net/STABLE/) does. What do you think about that?

I added the smarty assignments for the images because the rest of FD's code already does that and I thought it would be the way you guys do it :wink: If you want me to put the paths directly into the templates, just let me know.

I will wait for your response about the previous points before putting the copyright into a translatable string and changing the login screen title.

bilbo-the-hobbit commented:
Hello,

yes i agree everything should be done, to be themable and not on fixed path as we removed them long ago

Cheers

MCMic commented:
I think a centered logo would be ugly…
Can you provide screenshots of the result of your modifications for login page and recovery page?

In legacy theme setup_header was a green bar containing logo and version, but in breeze theme the designer decided to remove the bar and just have logo and version show up.
I’m not disturbed by having the logo in the corner.

I do think it’s better if you put the icons paths directly in the tpl (it also means you need to use `&amp;` instead of & to have HTML valid code).
The code putting paths in vars is a bit old.

maybe the copynotice should go in a separate tpl file as it will now be used in three places? (menu, login, recovery)

bilbo-the-hobbit commented:
Hello,

i opened the ticket https://forge.fusiondirectory.org/issues/5671 to help integrating those changes in our code

Cheers

MCMic commented:
I pushed it and tried to adapt it a bit but I don’t like the result.
The copyright notice end up at the top of the screen, is that on purpose?

MCMic commented:
Merged and adapted, can be tested on http://demo-dev.fusiondirectory.org/
(Deleted previous comment, I was wrong it was a local problem on my machine)

dasJ commented:
Okay, I reworked the entire PR. I split my changed into individual commits and also worked your changes into the PR.

Changes:

- Remove references to gosa
- Localize the full copyright and put it into an own template
- Include this copyright into the login and recovery screens
- Change the Login title to "Login"
- Redesign the password recovery step 4
- Allow going back from recovery to login

I created multiple variants with the logo at different positions, the following are screenshots

# How it looks on master

![master login](https://user-images.githubusercontent.com/4971975/28490891-684c5fa4-6ee5-11e7-9671-3e2ce8a1b4d8.png)

![master recovery](https://user-images.githubusercontent.com/4971975/28490918-e614d79a-6ee5-11e7-99f6-0c0fe866b119.png)

# Variant A

This is the state of this PR, without any logo and version

![a login](https://user-images.githubusercontent.com/4971975/28490922-076b0e28-6ee6-11e7-9535-fe54f4d37efb.png)

![a recovery](https://user-images.githubusercontent.com/4971975/28490925-1082712c-6ee6-11e7-964b-05b49d108a8c.png)

# Variant B

This has the (IMO oddly placed) logo top left, but no version number on the right

![b login](https://user-images.githubusercontent.com/4971975/28490929-2c8daf62-6ee6-11e7-83af-c3308ef83a04.png)

![b recovery](https://user-images.githubusercontent.com/4971975/28490930-31d08846-6ee6-11e7-84dd-4f3992574f79.png)

# Variant C

I thought "Why not add the icon to the header of the container?"

![c login](https://user-images.githubusercontent.com/4971975/28490937-4939d1cc-6ee6-11e7-8275-ec0c32012fc1.png)

![c recovery](https://user-images.githubusercontent.com/4971975/28490939-4ff6a224-6ee6-11e7-9a84-e150edc9d93d.png)

---

I hope these screenshots help you decide, maybe even bring new ideas.

bilbo-the-hobbit commented:
hello,

i really like version C, looks more professional to me, we will look to integrate those changes

Thanks !

kadogo commented:
Hello, I like the C too ^^

MCMic commented:
We cannot use «Login» in the end because the translation will be wrong (same word in different places will be translated in the same way so it will be translated the same as «login» in the user edition).
So I guess we’re going back to «login screen»

MCMic commented:
It seems you made the PR against master again but the dev branch is 1.3-dev right now and it already contains the version B.
Could you make a PR against 1.3-dev which contains the changes which were not merged yet? It seems some of you commits contain both merged stuff and new stuff (like the gosa removing one)

dasJ commented:
I limited the PR to the changes not already merged. I still rename the gosa CSS class. I also changed the login title to "FusionDirectory Login" if this is okay for @MCMic . If not, I can still rebase the commit out.

MCMic commented:
I would avoid putting «FusionDirectory» in the title as it is already written in the logo.
I think «Sign in» would work fine it’s already the text in the button so it should be translated fine.

Can you explain why you use the fd-logo id for the icon in the recovery page? And why you change the size from 48 to 30px?

dasJ commented:
I replaced the "Login" with "Sign in" now.

@MCMic As I removed the logo on top from the recovery page, I also added it to the container heading.
However, this looked awkward next to the keys icon.
I also shrinked the logo a bit because it seemed too big.

MCMic commented:
Thanks, this is now merged in 1.3-dev.
I fixed back the legacy theme so that it looks like before for nostalgics.
